### PR TITLE
workflows: tweaks according to decision

### PR DESCRIPTION
**Pull request by gsilvapt**
Only runs test workflows on pull requests to develop branch and when changes are merged into main.
Publish workflows are already as we discussed.

Ref: https://github.com/surface-security/surface/issues/86

---

I'm simultaneously changing the repositories settings in regards to protected branches and tags.

### CODE REVIEW

**Comment by gsilvapt** [comment]
Only @surface-security/core can push to `main` in this repository now, so we have better control over who is allowed to create releases (and push artifacts to pypi).
image

**Comment by bogdanoniga** [comment]
[CODEOWNERS file](https://github.com/surface-security/django-kubernetes/blob/main/.github/CODEOWNERS) can be used & enable `Require review from Code Owners`
image

**Comment by gsilvapt** [comment]
> [CODEOWNERS file](https://github.com/surface-security/django-kubernetes/blob/main/.github/CODEOWNERS) can be used & enable `Require review from Code Owners`
> 
> image

Thanks! `CODEOWNERS` is worth it for PR reviews and indeed we should migrate some time in the future. However, `CODEOWNERS` does not apply to who can push to protected branches, tags nor create releases in the repository, which is what this issue is covering.
I honestly need to find a way to propagate these type of changes quickly across the entire org, because going 1 by 1 is cumbersome.

On a side note: I believe your codeowners should be @surface-security/core and not the other team. The other team is more as a "thank you note" for everyone that has contributed in the past, whereas the core is for the core group of maintainers.

**Comment by fopina** [comment]
Integrations from develop to main are meant to do via PR? If so, tests could there instead of “push: main”

**Comment by gsilvapt** [comment]
> Integrations from develop to main are meant to do via PR? If so, tests could there instead of “push: main”

Although more work, I think it's better because we would be running the test suite on top of all changes from `develop` to `main` and have a clear gateway to observe issues and/or incompatible changes.

And I agree with you, I dislike running tests just because. We could run the test suite when things get merged into `main` but feels unnecessary if they already run when the PR is created.

Updated.

**Review by fopina — APPROVED** [review]

**Pull request merged**
